### PR TITLE
fix(test): convert editor tests to Kotlin

### PR DESCRIPTION
**Pull request by djansen-redhat**
JIRA issue URL: n/a

Convert the editor functional tests to Kotlin

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

### CODE REVIEW

File: server/functional-test/src/test/java/org/zanata/feature/editor/EditorFilterMessagesTest.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    @Test(timeout = MAX_SHORT_TEST_DURATION.toLong())
+    fun editorFilterIsBookmarkable() {
+        val urlForEditor = String.format(BasicWorkFlow.EDITOR_TEMPLATE, "about-fedora",
+                "master", "fr", document)
+        val urlWithFilterCondition = "$urlForEditor;search:hello%20w;resid:res1"
+        val editorPage = BasicWorkFlow().goToPage(urlWithFilterCondition,
+                EditorPage::class.java)
+
+        assertThat(editorPage.messageSources).containsExactly("hello world")
+        assertThat(editorPage.filterQuery.trim { it <= ' ' })
```
Comment (seanf) [inline server/functional-test/src/test/java/org/zanata/feature/editor/EditorFilterMessagesTest.kt]: trim what?

File: server/functional-test/src/test/java/org/zanata/feature/editor/TranslateTextTest.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    private val testFileGenerator = TestFileGenerator()
+
+    @Before
+    fun before() {
+        zanataRestCaller = ZanataRestCaller()
+        BasicWorkFlow().goToHome().deleteCookiesAndRefresh()
+        assumeFalse(
+                "",
+                File(CleanDocumentStorageRule.getDocumentStoragePath() +
+                        "${File.separator}documents${File.separator}").exists())
```
Comment (seanf) [inline server/functional-test/src/test/java/org/zanata/feature/editor/TranslateTextTest.kt]: I know this is probably pre-existing logic, but wouldn't it be better just to delete the the directory, or simply fail the test? 

If this condition fails, I think the test will be skipped with little or no warning. That doesn't seem right.

Comment (djansen-redhat) [reply server/functional-test/src/test/java/org/zanata/feature/editor/TranslateTextTest.kt]: Indeed, it's unnecessary - the rule throws if a delete fails.

File: server/functional-test/src/test/java/org/zanata/feature/editor/EditorFilterMessagesTest.kt (hunk 1 above)
Comment (djansen-redhat) [reply server/functional-test/src/test/java/org/zanata/feature/editor/EditorFilterMessagesTest.kt]: It actually trims the filter query, which adds a space on the end - but I've changed it to test against the _actual_ string. Good catch.

Comment (seanf) [reply server/functional-test/src/test/java/org/zanata/feature/editor/EditorFilterMessagesTest.kt]: I'm not sure I really caught anything. I was trying to work out what exactly it was doing. Was it trimming any characters "less than or equal to" SPACE, ie control characters?

**Comment by djansen-redhat** [comment]
Part of https://github.com/zanata/zanata-platform/pull/946

**Pull request closed without merging**